src/sponsorabilityFinder.ts

```diff
@@ -0,0 +1,300 @@
+            withCountsAndRepoID = this.addContributionCount(sponsorables, contributionCountOfUser);
+
+            const storeSponsorableContributors: SponsorableWithContributionCount[] = [];
+            withCountsAndRepoID.forEach((objectWithContributorData) => {
+                if (key === objectWithContributorData.repoId) {
+                    storeSponsorableContributors.push(...[objectWithContributorData.contributor]);
+                    withContributionCounts.set(
+                        objectWithContributorData.repoId,
+                        storeSponsorableContributors
+                    );
+                }
+            });
+        }
+
+        return withContributionCounts;
+    }
+
+    public addContributionCount(
+        sponsorableContributor: Map<string, Sponsor[]>,
+        contributors: ContributionCountOfUserIntoRepo[]
+    ): {
+        repoId: string;
+        contributor: SponsorableWithContributionCount;
+    }[] {
+        const allSponsorableWithCount: {
+            repoId: string;
+            contributor: SponsorableWithContributionCount;
+        }[] = [];
+
+        for (const [key, sponsors] of sponsorableContributor) {
+            sponsors.forEach((sponsorable) => {
+                const contributionsCount = this.getContributionCountOfUser(
+                    sponsorable.login,
+                    contributors
+                );
+
+                const withContributionCount: SponsorableWithContributionCount =
+                    this.convertToSponsorableWithCounts(sponsorable, contributionsCount);
+
+                allSponsorableWithCount.push({ repoId: key, contributor: withContributionCount });
+            });
+        }
+
+        return allSponsorableWithCount;
+    }
+
+    public convertToSponsorableWithCounts(
+        sponsorable: Sponsor,
+        contributionsCount: number
+    ): SponsorableWithContributionCount {
+        const withContributionCount: SponsorableWithContributionCount = {
+            type: sponsorable.__typename,
+            email: sponsorable.email ?? '',
+            login: sponsorable.login,
+            url: `https://github.com/${sponsorable.login}`,
+            sponsorListingName: sponsorable.sponsorsListing.name ?? '',
+            sponsorsLink: sponsorable.sponsorsListing.dashboard ?? '',
+            contributionsCount: contributionsCount ?? 0,
+        };
+
+        return withContributionCount;
+    }
+
+    public getContributionCountOfUser(
+        userLogin: string,
+        contributors: ContributionCountOfUserIntoRepo[]
+    ): number {
+        let contributionCounts = 0;
+
+        contributors.forEach((contributor) => {
+            if (userLogin === contributor.login) {
+                contributionCounts = contributor.contributions;
+            }
+        });
+
+        return contributionCounts;
+    }
+
+    /*
+
+    for each repo count the number of contributions that were made,
```

Comment:
More of this kind of comment might be helpful while the naming and logic is still being worked out. Then, by the end, hopefully the code will be clear enough that the comments will be redundant and can be deleted.

Reply:
yes, planning to remove all comments once my logic and naming is fleshed out